src/kbucket.rs

```diff
@@ -1,146 +1,148 @@
 #![allow(unused)]
 
-use std::string::String;
+use crate::helper::{Identifier, Node, U256};
+use std::collections::HashMap;
 use uint::*;
-use crate::helper::{Identifier, 
-    Node,
-    U256,
-};
 
 const BUCKET_SIZE: usize = 20;
 const MAX_BUCKETS: usize = 256;
 
 type Bucket = [Option<Node>; BUCKET_SIZE];
 
-pub enum StoreValue {
-    Node(Node),
-    Sample(String),  // Define a sample, and change the type to a sample
+pub enum FindNodeResult {
+    // I don't think this should be "Option<T>".  Fix later
+    Found(Option<Node>),
+    NotFound(Vec<Option<Node>>),
+}
+pub struct SearchResult {
+    pub found: bool,
+    pub bucket_index: usize,
+    pub column_index: usize,
 }
```

Comment:
you could just make this `Option<NodeRef>` or something where `type NodeRef = (usize, usize)`

not clear who needs these indices outside this module so you may want more expressive typing that just a pair of `usize`...